howso/get_cases.amlg

```diff
@@ -286,26 +286,80 @@
 		)
 	)
 
+
 	;return all cases that match the specified condition
 	;parameters:
-	; precision: flag, whether to query for 'exact' matches; if set to 'similar' will query for similar values intead of 'exact'
-	; condition_session: optional, if specified ignores condition and returns cases for the specified session id
-	; condition: assoc of feature->value(s)
+	; condition_session: optional, if specified ignores 'condition' and 'condition_filter_query' and returns cases for the specified session id
+	; condition_filter_query: optional, list of query conditions. If specified ignores 'condition' and returns cases for the specified fitery query
+	; condition: optional assoc of feature->value(s). Condition for the returned cases. If 'condition_filter_query' is provided, then this parameter is ignored.
 	;	no value = must have feature
 	;	- for continuous or numeric ordinal features:
 	;	one value = must equal exactly the value or be close to it for fuzzy match
 	;	two values = inclusive between
 	;	- for nominal or string ordinal features:
 	;	n values = must match any of these values exactly
-	; num_cases: optional, limit on the number of cases to get; If set to zero there will be no limit.
-	;		If null, will be set to k if precision is "similar" or no limit if precision is "exact". default is null
+	; precision: flag, default is "exact". Whether to query for 'exact' matches when using the 'condition' parameter; if set to 'similar' will query for similar values intead of 'exact'
+	;	If 'condition_filter_query' is provided, then this parameter is ignored.
+	; num_cases: optional, limit on the number of cases to get when using the 'condition' parameter. Works with or without 'condition_filter_query'. default is null.
+	;	If 'condition_filter_query' is specified:
+	;		If null there will be no limit.
+	;	If 'condition_filter_query' is not specified:
+	;		If null, will be set to k if precision is "similar" or no limit if precision is "exact".
 	#!GetCasesByCondition
 	(declare (assoc
-			condition (assoc)
-			condition_session (null)
-			precision "exact"
-			num_cases (null)
+		condition (assoc)
+		condition_filter_query (list)
+		condition_session (null)
+		precision "exact"
+		num_cases (null)
+		)
+	
+		(if (!= (null) condition_session)
+			;get all the case ids in the order they were stored for the specified session
+			(retrieve_from_entity condition_session ".replay_steps")
+			(seq
```

Comment:
I'd prefer a newline between the `(retrieve_from_entity)` and the `(seq`. It's hard to tell that these are the if/else branches without it